lib/grpc/client/adapters/mint/connection_process/connection_process.ex

```diff
@@ -0,0 +1,177 @@
+defmodule GRPC.Client.Adapters.Mint.ConnectionProcess do
+  use GenServer
+
+  alias GRPC.Client.Adapters.Mint.ConnectionProcess.State
+  alias GRPC.Client.Adapters.Mint.StreamResponseProcess
+
+  require Logger
+
+  def start_link(scheme, host, port, opts \\ []) do
+    GenServer.start_link(__MODULE__, {scheme, host, port, opts})
+  end
+
+  def request(pid, method, path, headers, body, opts \\ []) do
+    GenServer.call(pid, {:request, method, path, headers, body, opts})
+  end
+
+  def disconnect(pid) do
+    GenServer.call(pid, {:disconnect, :brutal})
+  end
+
+  def stream_request_body(pid, request_ref, body) do
+    GenServer.call(pid, {:stream_body, request_ref, body})
+  end
+
+  ## Callbacks
+
+  @impl true
+  def init({scheme, host, port, opts}) do
+    # The current behavior in gun is return error if the connection wasn't successful
+    # Should we do the same here?
```

Comment:
We can design a separate UX if we want. What would be the alternative?

Reply:
Currently I'm spawning the connection async for mint. So, if the connection doesn't succeeds I need to perform a request to know that it wasn't successful. 

To match the behavior we have for GUN I would need to make connection spawn be sync OR create a `call` for the gen_server to check if the connection was successful